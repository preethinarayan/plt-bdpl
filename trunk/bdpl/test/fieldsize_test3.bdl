
struct Outer
{
	byte fz (fieldsize 7) satisfies{1} then{print(fz);};  // this will contain the value 6 after reading file
	struct A
	{
		struct C
		{
			byte bb (fieldsize fz); // fieldsize = 6
		}[fz]xx;
	
	}[*] a (fieldsize 79); // fieldsize is in bits !
}bb;

file "x:/test/file_test.dat" inp_file;
read(inp_file,bb);
print(bb);

