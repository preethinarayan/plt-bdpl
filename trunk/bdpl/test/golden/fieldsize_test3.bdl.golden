0x6struct:Outer
{
	fz => 0x6,
	a => ARRAY of struct:A
	[
		0 => struct:A
		{
			xx => ARRAY of struct:C
			[
				0 => struct:C
				{
					bb => 0x28,
				}
				,
				1 => struct:C
				{
					bb => 0xa,
				}
				,
				2 => struct:C
				{
					bb => 0x4,
				}
				,
				3 => struct:C
				{
					bb => 0x21,
				}
				,
				4 => struct:C
				{
					bb => 0x28,
				}
				,
				5 => struct:C
				{
					bb => 0x22,
				}
				,
			]
			,
		}
		,
		1 => struct:A
		{
			xx => ARRAY of struct:C
			[
				0 => struct:C
				{
					bb => 0xa,
				}
				,
				1 => struct:C
				{
					bb => 0x23,
				}
				,
				2 => struct:C
				{
					bb => 0x8,
				}
				,
				3 => struct:C
				{
					bb => 0x3a,
				}
				,
				4 => struct:C
				{
					bb => 0x10,
				}
				,
				5 => struct:C
				{
					bb => 0x24,
				}
				,
			]
			,
		}
		,
		2 => struct:A
		{
			xx => ARRAY of struct:C
			[
				0 => struct:C
				{
					bb => 0x29,
				}
				,
				1 => struct:C
				{
					bb => 0x0,
				}
				,
				2 => struct:C
				{
					bb => 0x0,
				}
				,
				3 => struct:C
				{
					bb => 0x0,
				}
				,
				4 => struct:C
				{
					bb => 0x0,
				}
				,
				5 => struct:C
				{
					bb => 0x0,
				}
				,
			]
			,
		}
		,
		3 => struct:A
		{
			xx => ARRAY of struct:C
			[
				0 => struct:C
				{
					bb => 0x0,
				}
				,
				1 => struct:C
				{
					bb => 0x0,
				}
				,
				2 => struct:C
				{
					bb => 0x0,
				}
				,
				3 => struct:C
				{
					bb => 0x0,
				}
				,
				4 => struct:C
				{
					bb => 0x0,
				}
				,
				5 => struct:C
				{
					bb => 0x0,
				}
				,
			]
			,
		}
		,
	]
	,
}
